observability/WFLY-19846-promote-mp-tel-2-to-default.adoc

```diff
@@ -0,0 +1,115 @@
+feature-team:
+ developer: jasondlee
+ sme:
+  -
+ outside-perspective:
+  -
+promotes: https://issues.redhat.com/browse/WFLY-19590
+promoted-by: https://github.com/wildfly/wildfly-proposals/issues/619
+---
+= [Default] Promote MicroProfile Telemetry 2.0 to WildFly Standard
+:author:            Jason Lee
+:email:             jasondlee@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+MicroProfile Telemetry 2.0 shipped with WildFly Preview 34. To meet WildFly and product roadmap goals, this feature needs to be migrated to the WildFly feature pack, and its stability level promoted to `DEFAULT`.
+
+=== User Stories
+
+User wants to deploy an application using MicroProfile Telemetry 2.0 from MicroProfile Platform 7.0.
+
+== Issue Metadata
+
+- https://issues.redhat.com/browse/WFLY-19846[]
+
+=== Related Issues
+
+- https://issues.redhat.com/browse/EAP7-2227[]
+
+=== Affected Projects or Components
+
+- MicroProfile Fault Tolerance
+- AMQP
+- Kafka
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+
+* Traditional standalone server (unzipped or provisioned by Galleon)
+* OpenShift Source-to-Image (S2I)
+* Bootable jar
+
+== Requirements
+
+* WildFly must fully support MicroProfile Telemetry 2.0:
+** Tracing - Existing support
+** Metrics - New feature
+** Logs - New feature
+* OpenTelemetry versions must be updated to match those required by MicroProfile Telemetry 2.0
+
+=== Changed requirements
+
+N/A
+
+=== Non-Requirements
+
+Currently, neither OpenTelemetry nor MicroProfile Telemetry are supported in domain mode. While this is now technically feasible, given the push nature of the export process, support for this will have to be delivered under a different (to-be-file) feature request.
+
+=== Future Work
+
+N/A
+
+== Backwards Compatibility
+
+* There are no known breaking changes. The upstream OpenTelemetry project made a number of breaking changes to the metrics semantic conventions (which could affect user dashboards), but metrics support will be new with this release, so those changes will not affect existing installations.
+
+=== Default Configuration
+
+* The default configuration will be unchanged and continue to work as is.
+
+=== Deployments
+
+Deployment behavior will be unaffected.
+
+=== Interoperability
+
+N/A
+
+== Implementation Plan
+
+N/A. Will be delivered in one release.
+
+== Admin Clients
+
+N/A
+
+== Security Considerations
+
+This change will have no effect on security. As stated in the MicroProfile Telemetry 1.0, there is no baked in notion of security with OpenTelemetry, so systems administrators will need to make sure that the endpoint for the push publication is properly configured, and that the receiving system -- whatever it may, e.g. and OpenTelemetry Collector -- is secured as desired and appropriate for that system. Such a configuration is outside the scope of the work here.
+
+[[test_plan]]
+== Test Plan
```

Comment:
Can you please add "Manual tests", "Miscellaneous checks" and "Compatibility tests" sections here according to the template? https://github.com/wildfly/wildfly-proposals/blob/790b15b89999850f151e25190266a44d8e16ab35/design-doc-template.adoc?plain=1#L145

Reply:
If you have something specific you'd like to see add, please describe that for me. I removed the section as I didn't see anything relevant. Perhaps different eyes will, of course, but these sections were deemed superfluous and removed.

Reply:
Thank you for your feedback! AFAIU user can use default WF Metrics, MP Telemetry metrics, MicroMeter metrics. But AFAIU it's not recommended to use more then one metrics together. What about to add manual check whether the server prints proper warning if more metrics systems are use together? My understanding is that such situation is not recommended based on [WF admin guide](https://docs.wildfly.org/34/Admin_Guide.html#Micrometer_Metrics): 

> Note that this an alternative to the existing WildFly Metrics extension. While they may be run concurrently, it is not advisable, as this will likely have an impact on server performance due to the duplicated metrics collection.

Reply:
@marekkopecky We currently document that running multiple is not wise, and I think that's sufficient for now. We can certainly add a warning, but that's not just a single line change here or there, so, if we want to do that, that will have to be an RFE in a future release.

Reply:
Thank you for your feedback! I understand your PoV, but I should mainly take care about the quality, so I created an issue for this, so it can be triaged separately.

Reply:
That's fine. As I noted, it's not a simple, one line change, and it will affect multiple modules, so that change should take some more time to make sure it's done correctly, and definitely should done under another JIRA. Thanks for filing that.

Reply:
@jasondlee What do you mean by RFE? A Feature Request for a new feature at default stability? Or do you mean an Enhancement, not involving any user controllable behavior?

Reply:
> @jasondlee What do you mean by RFE? A Feature Request for a new feature at default stability? Or do you mean an Enhancement, not involving any user controllable behavior?

@bstansberry Sorry. I meant an Enhancement.